gnomad_constraint/plots/generate_constraint_plots.R

```diff
@@ -116,8 +121,68 @@ lof_upper_bin <- list(
   v4 = "lof.oe_ci.upper_bin_decile"
 )
 
+preferred_transcripts <- list(
+  v2 = "canonical",
+  v4 = "mane_select"
+)
+
+
+filter_transcripts <- function(
+    df,
+    version,
+    preferred_transcripts_only = TRUE,
+    enst_only = TRUE,
+    include_flagged_transcripts = FALSE) {
+  # Filter transcripts to specified criteria
+  # df: Dataframe on which filters should be applied
+  # version: Version of the dataset, either 'v2' or 'v4. Dataframe will be filtered to
+  # variants in this version.
+  # preferred_transcripts_only: Whether to filter to only the preferred transcript.
+  # Default is TRUE. For 'v4', 'mane_select' are the preferred transcripts. For 'v2',
+  # 'canonical' transcripts are the preferred transcripts.
+  # enst_only: Whether to filter to only Ensembl transcripts. Default is TRUE.
+  # include_flagged_transcripts: Whether to keep transcripts with a constraint flag.
+  # Default is FALSE.
+  # Returns: Filtered dataframe
+
+  # Filter to transcripts present in specified version
+  df <- filter(df, !!sym(version))
+
+  # Filter to only Ensembl transcripts if specified
+  if (enst_only) {
+    df <- filter(df, grepl("^ENST", .data$transcript))
+  }
+
+  # Filter to exclude flagged transcripts if specified
+  if (!include_flagged_transcripts) {
+    if (version == "v4") {
+      df <- filter(df, .data$constraint_flags == "[]")
+    }
+    if (version == "v2") {
+      df <- filter(df, .data$constraint_flag == "")
+    }
+  }
+
+  # Filter to only preferred transcripts if specified
+  if (preferred_transcripts_only == TRUE) {
+    df <- filter(df, !!sym(preferred_transcripts[[version]]) == "true")
+  }
+
+  return(df)
+}
+
 for (version in versions_to_plot) {
-  gene_list_sums <- summarize_gene_lists(gene_data, lof_upper_bin[[version]], version)
+  filtered_data <- filter_transcripts(
+    gene_data,
+    version,
+    preferred_transcripts_only = TRUE,
+    enst_only = TRUE,
+    include_flagged_transcripts = FALSE
+  )
```

Comment:
Those are the defaults 
```suggestion
  filtered_data <- filter_transcripts(gene_data, version)
```